### PR TITLE
Added tmp102 temperature sensor support

### PR DESCRIPTION
**Pull request by timsavage**
## Description:
Added support for the TMP102 temperature sensor. 

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#451

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by timsavage** [comment]
Not sure why Travis is still pending. All builds have completed successfully.

**Comment by glmnet** [comment]
> Not sure why Travis is still pending. All builds have completed successfully.

It sometimes just can't tell GitHub the outcome, I fix it by re triggering the build, but I believe you need some kind of permission for that

File: esphome/components/tmp102/tmp102.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ESP_LOGE(TAG, "Communication with TMP102 failed!");
+  }
+  LOG_UPDATE_INTERVAL(this);
+  LOG_SENSOR("  ", "Temperature", this);
+}
+
+void TMP102Component::update() {
+  uint16_t raw_temperature;
+  if (!this->read_byte_16(TMP102_REGISTER_TEMPERATURE, &raw_temperature, 50)) {
+    this->status_set_warning();
```
Comment (glmnet) [inline esphome/components/tmp102/tmp102.cpp:32]: if you ever call set_warning you should call status_clear_warning at the end

File: esphome/components/tmp102/sensor.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+def to_code(config):
+    var = cg.new_Pvariable(config[CONF_ID])
+    yield cg.register_component(var, config)
+    yield i2c.register_i2c_device(var, config)
+    yield sensor.register_sensor(var, config)
+
+    if CONF_TEMPERATURE in config:
+        sens = yield sensor.new_sensor(config[CONF_TEMPERATURE])
+        cg.add(var.set_temperature(sens))
```
Comment (glmnet) [inline esphome/components/tmp102/sensor.py:34]: this is not needed

**Review by glmnet — CHANGES_REQUESTED** [review]

File: esphome/components/tmp102/tmp102.cpp (hunk 1 above)
Comment (jesserockz) [reply esphome/components/tmp102/tmp102.cpp:32]: @timsavage Please follow up on this. Either remove the `status_set_warning` and just create a log with `ESP_LOGW` or clear the status once it is working.

Comment (timsavage) [reply esphome/components/tmp102/tmp102.cpp:32]: I'll be free to have a look mid October.

File: esphome/components/tmp102/tmp102.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    return;
+  }
+
+  raw_temperature = raw_temperature >> 4;
+  float temperature = raw_temperature * TMP102_CONVERSION_FACTOR;
+  ESP_LOGD(TAG, "Got Temperature=%.1f°C", temperature);
+
+  this->publish_state(temperature);
+}
```
Comment (jesserockz) [inline esphome/components/tmp102/tmp102.cpp:41]:
```suggestion
  this->publish_state(temperature);
  this->status_clear_warning();
}
```

**Review by glmnet — APPROVED** [review]
LGTM! docs too

**Comment by timsavage** [comment]
Change looks good.

**Pull request merged**